save.go

```diff
@@ -354,14 +354,35 @@ func copyFile(dst, src string) error {
 }
 
 func copyWithoutImportComment(w io.Writer, r io.Reader) error {
-	sc := bufio.NewScanner(r)
-	for sc.Scan() {
-		_, err := w.Write(append(stripImportComment(sc.Bytes()), '\n'))
+	var eof bool
+	b := bufio.NewReader(r)
+	for {
+		l, err := b.ReadBytes('\n')
 		if err != nil {
-			return err
+			if err == io.EOF {
+				eof = true
+			} else {
+				return err
+			}
+		}
+
+		// If we have data then write it out...
+		if len(l) > 0 {
+			// Trim the \n if it exists
+			if l[len(l)-1] == '\n' {
+				l = l[0 : len(l)-1]
+			}
+
+			if _, err := w.Write(append(stripImportComment(l), '\n')); err != nil {
```

Comment:
Since the call to w.Write is made primarily for its side effect, IMO it doesn't belong inside the if-stmt, it should go on its own line above.